mockall_derive/src/mock_trait.rs

```diff
@@ -176,3 +174,45 @@ impl MockTrait {
         )
     }
 }
+
+fn write_join_path(
+    output: &mut String,
+    path: &Path,
+    sep: &str,
+) -> fmt::Result {
+    let mut is_first = true;
+    for segment in &path.segments {
+        if is_first {
+            is_first = false;
+            write!(output, "{}", segment.ident)?;
+        } else {
+            write!(output, "{}{}", sep, segment.ident)?;
+        }
+    }
+    Ok(())
+}
+
+fn hash_path_arguments(path: &Path) -> Option<u64> {
```

Comment:
I don't have an example. This was done since I assumed processing the hash + concatenating the ident are symmetric. Whereas before both were done on the last segment, now we go over all the segments. Happy to revert this part to only process the last segment.